File: client/src/schemas/task-card.ts
import { z } from 'zod'

export const createTaskCardSchema = z.object({
    title: z.string()
        .min(1, { message: "Title is required" }),
    description: z.string(),
    listId: z.string().uuid(),
    position: z.number().int()
})

export const taskCardResponseSchema = z.object({
    id: z.string(),
    title: z.string(),
    description: z.string(),
    listId: z.string(),
    position: z.number(),
    createdAt: z.string(),
    updatedAt: z.string(),
    priorityId: z.string(),
    statusId: z.string(),
    dueDate: z.string().optional(),
    status: z.object({
        id: z.string().uuid(),
        name: z.string()
    }).nullable(),
    activities: z.array(z.object({
        id: z.string().uuid().optional(),
        description: z.string()
    })).nullable(),
    labels: z.array(z.object({
        id: z.string().uuid(),
        name: z.string()
    })).nullable()
})

export type CreateTaskCardData = z.infer<typeof createTaskCardSchema>
export type TaskCardResponse = z.infer<typeof taskCardResponseSchema>

File: server/PMS_Project/PMS_Project.Domain/src/Models/TaskCard.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PMS_Project.Domain.Models
{
    public class TaskCard : BaseClass
    {
        [Required]
        [MaxLength(500)]
        public string Description { get; set; }

        [Required]
        public Guid ListId { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        [Required]
        public string Title { get; set; }

        [Required]
        public Guid PriorityId { get; set; }

        public string? DueDate { get; set; }

        [Required]
        public Guid StatusId { get; set; }

        public int Position { get; set; }

        

        [ForeignKey(nameof(ListId))]
        public AppList List { get; set; } //'List' is a reserved keyword in C# so we use 'AppList' instead

        [ForeignKey(nameof(StatusId))]
        public Status? Status { get; set; } // Navigation property

        [ForeignKey(nameof(PriorityId))]
        public Priority? Priority { get; set; } // Navigation property

        public ICollection<TaskCard_ProjectBoardLabel>? TaskCard_ProjectBoardLabels { get; set; } // Navigation property
        public ICollection<TaskCard_User>? TaskCard_Users { get; set; } // Navigation property for many-to-many relationship with User
        public ICollection<TaskCardActivity>? Activities { get; set; } // Navigation property for one-to-many relationship
        public ICollection<TaskCardChecklist>? Checklists { get; set; } // Navigation property for one-to-many relationship
    }
}